### ISSUE TITLE
Handling nested data types in ANSI dialect

### ISSUE THREAD

**Issue by courentin** [sqlfluff/sqlfluff#1732]
When trying to lint this query:
```SQL
SELECT CAST(1 AS ARRAY<INT>)
FROM table
```
I have a parsing error.

I'm not sure of the `<>` syntax but at least it appear [here](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#array_type) and in Presto (which [should work](https://prestodb.io/docs/current/overview/concepts.html#statement) with ansi compatible queries).

Note that using parentheses instead of chevrons does not produce parsing error. However, the keywork `identifier` is recognized as a column identifier (which produces a `L014` lint error).

<details>
<summary>See parsing log when using parenthesis</summary>

```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'SELECT'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    function:
[L:  1, P:  8]      |                        function_name:
[L:  1, P:  8]      |                            function_name_identifier:         'CAST'
[L:  1, P: 12]      |                        bracketed:
[L:  1, P: 12]      |                            start_bracket:                    '('
[L:  1, P: 13]      |                            [META] indent:
[L:  1, P: 13]      |                            expression:
[L:  1, P: 13]      |                                literal:                      '1'
[L:  1, P: 14]      |                            whitespace:                       ' '
[L:  1, P: 15]      |                            keyword:                          'AS'
[L:  1, P: 17]      |                            whitespace:                       ' '
[L:  1, P: 18]      |                            data_type:
[L:  1, P: 18]      |                                data_type_identifier:         'ARRAY'
[L:  1, P: 23]      |                                bracketed:
[L:  1, P: 23]      |                                    start_bracket:            '('
[L:  1, P: 24]      |                                    [META] indent:
[L:  1, P: 24]      |                                    expression:
[L:  1, P: 24]      |                                        column_reference:
[L:  1, P: 24]      |                                            identifier:       'INT'
[L:  1, P: 27]      |                                    [META] dedent:
[L:  1, P: 27]      |                                    end_bracket:              ')'
[L:  1, P: 28]      |                            [META] dedent:
[L:  1, P: 28]      |                            end_bracket:                      ')'
[L:  1, P: 29]      |            newline:                                          '\n'
[L:  2, P:  1]      |            [META] dedent:
[L:  2, P:  1]      |            from_clause:
[L:  2, P:  1]      |                keyword:                                      'FROM'
[L:  2, P:  5]      |                whitespace:                                   ' '
[L:  2, P:  6]      |                from_expression:
[L:  2, P:  6]      |                    [META] indent:
[L:  2, P:  6]      |                    from_expression_element:
[L:  2, P:  6]      |                        table_expression:
[L:  2, P:  6]      |                            table_reference:
[L:  2, P:  6]      |                                identifier:                   'table'
[L:  2, P: 11]      |                    [META] dedent:
[L:  2, P: 11]      |    newline:                                                  '\n'
```

</details>

## Expected Behaviour

No parsing error

## Observed Behaviour

Got:
```
Found unparsable section: '<INT>'
````

<details>
<summary>Parsing result</summary>

```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'SELECT'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    function:
[L:  1, P:  8]      |                        function_name:
[L:  1, P:  8]      |                            function_name_identifier:         'CAST'
[L:  1, P: 12]      |                        bracketed:
[L:  1, P: 12]      |                            start_bracket:                    '('
[L:  1, P: 13]      |                            [META] indent:
[L:  1, P: 13]      |                            expression:
[L:  1, P: 13]      |                                literal:                      '1'
[L:  1, P: 14]      |                            whitespace:                       ' '
[L:  1, P: 15]      |                            keyword:                          'AS'
[L:  1, P: 17]      |                            whitespace:                       ' '
[L:  1, P: 18]      |                            data_type:
[L:  1, P: 18]      |                                data_type_identifier:         'ARRAY'
[L:  1, P: 23]      |                            unparsable:                       !! Expected: 'Nothing...'
[L:  1, P: 23]      |                                raw:                          '<'
[L:  1, P: 24]      |                                raw:                          'INT'
[L:  1, P: 27]      |                                raw:                          '>'
[L:  1, P: 28]      |                            [META] dedent:
[L:  1, P: 28]      |                            end_bracket:                      ')'
[L:  1, P: 29]      |            newline:                                          '\n'
[L:  2, P:  1]      |            [META] dedent:
[L:  2, P:  1]      |            from_clause:
[L:  2, P:  1]      |                keyword:                                      'FROM'
[L:  2, P:  5]      |                whitespace:                                   ' '
[L:  2, P:  6]      |                from_expression:
[L:  2, P:  6]      |                    [META] indent:
[L:  2, P:  6]      |                    from_expression_element:
[L:  2, P:  6]      |                        table_expression:
[L:  2, P:  6]      |                            table_reference:
[L:  2, P:  6]      |                                identifier:                   'table'
[L:  2, P: 11]      |                    [META] dedent:
[L:  2, P: 11]      |    newline:                                                  '\n'

==== parsing violations ====
L:   1 | P:  23 |  PRS | Line 1, Position 23: Found unparsable section: '<INT>'
WARNING: Parsing errors found and dialect is set to 'ansi'. Have you configured your dialect?
```

</details>

## Dialect

ANSI

## Version

0.6.9

## Configuration

Default one
